boards/arm/b_g474e_dpow1/board.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * Copyright (c) 2022 The Chromium OS Authors
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr.h>
+#include <device.h>
+#include <devicetree.h>
+#include <drivers/gpio.h>
+#include <drivers/adc.h>
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(board, LOG_LEVEL_DBG);
+
+#if DT_HAS_COMPAT_STATUS_OKAY(usbc_port)
+
+#define VBUS DT_PATH(vbus)
+#define LED_LABEL       DT_GPIO_LABEL(LED0_NODE, gpios)
+#define LED_PIN         DT_GPIO_PIN(LED0_NODE, gpios)
+#define LED_FLAGS       DT_GPIO_FLAGS(LED0_NODE, gpios)
+
+/* Common settings supported by most ADCs */
+#define ADC_RESOLUTION          12
+#define ADC_GAIN                ADC_GAIN_1
+#define ADC_REFERENCE           ADC_REF_INTERNAL
+#define ADC_ACQUISITION_TIME    ADC_ACQ_TIME_DEFAULT
+#define ADC_REF_MV              3300
```

Comment:
you can now use the recently added adc dt-spec facilities to make code more portable
https://github.com/zephyrproject-rtos/zephyr/pull/43030

and soon:
https://github.com/zephyrproject-rtos/zephyr/pull/46110

Reply:
Updated